src/ResourceManagement/BatchAI/BatchAIFileServersImpl.cs

```diff
@@ -46,37 +59,56 @@ protected override IBatchAIFileServer WrapModel(FileServerInner inner)
             {
                 return null;
             }
-            return new BatchAIFileServerImpl(inner.Name, inner, this.Manager);
+            return new BatchAIFileServerImpl(inner.Name, this.workspace, inner);
+        }
+        
+        public IEnumerable<IBatchAIFileServer> List()
+        {
+            return WrapList(Extensions.Synchronize(() => Inner.ListByWorkspaceAsync(workspace.ResourceGroupName, workspace.Name))
+                .AsContinuousCollection(link => Extensions.Synchronize(() => Inner.ListByWorkspaceNextAsync(link))));
+        }
+
+        public async Task<IPagedCollection<IBatchAIFileServer>> ListAsync(bool loadAllPages = true, CancellationToken cancellationToken = new CancellationToken())
+        {
+            return await PagedCollection<IBatchAIFileServer, FileServerInner>.LoadPage(
+                async (cancellation) => await Inner.ListByWorkspaceAsync(workspace.ResourceGroupName, workspace.Name, cancellationToken: cancellation),
+                Inner.ListByWorkspaceNextAsync,
+                WrapModel, loadAllPages, cancellationToken);
         }
 
-        protected override async Task<IPage<FileServerInner>> ListInnerAsync(CancellationToken cancellationToken)
+        public IBatchAIFileServer GetById(string id)
```

Comment:
why this one and `GetByIdAsync` just below it has different implementations?

Reply:
Fixed.